service/service.go

```diff
@@ -0,0 +1,311 @@
+			closestWeekday = slices.Min(closestWeekdaysInThisWeek)
+		} else {
+			//если следующий день по плану находится на следующей неделе
+			closestWeekday = slices.Min(wVals)
+		}
+
+		if closestWeekday < currentWeekday {
+			addDaysCnt = closestWeekday - (currentWeekday - 7)
+		} else {
+			addDaysCnt = closestWeekday - currentWeekday
+		}
+
+		newDate = currentDate.AddDate(0, 0, addDaysCnt)
+
+	case "m":
+		var mVals []int
+		needExactMonth := false
+		dVals := nextDateRequest.Repeat.Values[0]
+		if len(nextDateRequest.Repeat.Values) == 2 {
+			needExactMonth = true
+			mVals = nextDateRequest.Repeat.Values[1]
+		}
+
+		repeatDaysMap := make(map[int]bool, len(dVals))
+		repeatMonthsMap := make(map[int]bool, len(dVals))
+		hasLastDayRepeat := false
+		hasPreLastDayRepeat := false
+		for _, dVal := range dVals {
+			if dVal > 0 {
+				repeatDaysMap[dVal] = true
+			} else if dVal == -1 {
+				hasLastDayRepeat = true
+			} else if dVal == -2 {
+				hasPreLastDayRepeat = true
+			}
+		}
+		for _, mVal := range mVals {
+			repeatMonthsMap[mVal] = true
+		}
+
+		isCurrentDatePast := currentDate.Before(nowDate)
+
+		//Если дата где-то в прошлом, то сразу доведём до сегодняшней даты
+		if isCurrentDatePast {
+			currentDate = nowDate
+		}
+
+		newDay := currentDate.Day()
+		newMonth := int(currentDate.Month())
+
+		lastMonthDay, preLastMonthDay := get2LastMonthDays(currentDate)
+
+		newDate = currentDate
+		//Если дата задачи была в прошлом и сегодняшняя дата попадает под правило повторения
+		if isCurrentDatePast && (!needExactMonth || repeatMonthsMap[newMonth]) {
+			if hasPreLastDayRepeat && newDay == preLastMonthDay {
+				break
+			}
+			if hasLastDayRepeat && newDay == lastMonthDay {
+				break
+			}
+			if repeatDaysMap[newDay] {
+				break
+			}
+		}
+
+		for true {
+			newDate = newDate.AddDate(0, 0, 1)
+			if int(newDate.Month()) != newMonth {
+				lastMonthDay, preLastMonthDay = get2LastMonthDays(newDate)
+				newMonth = int(newDate.Month())
+			}
+
+			if !needExactMonth || repeatMonthsMap[newMonth] {
+				newDay = newDate.Day()
+				if hasPreLastDayRepeat && newDay == preLastMonthDay {
+					break
+				}
+				if hasLastDayRepeat && newDay == lastMonthDay {
+					break
+				}
+				if repeatDaysMap[newDay] {
+					break
+				}
+			}
+		}
+	case "y":
+		newDate = currentDate.AddDate(1, 0, 0)
+
+		for newDate.Before(nowDate) {
+			newDate = newDate.AddDate(1, 0, 0)
+		}
+	}
+
+	return newDate, nil
+}
+
+// AddTask добавляет задание
+func (s *Service) AddTask(addTaskRequest model.AddTaskRequest) (model.AddTaskResponse, error) {
+	now := time.Now()
+	nowDate := time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, time.Local)
+	taskDate := nowDate
+	// Если дата в запросе не указана, то сегодняшнюю берём
+	if addTaskRequest.Date != "" {
+		reqDate, err := DateParse(addTaskRequest.Date)
+		if err != nil {
+			return model.AddTaskResponse{}, fmt.Errorf("ошибка парсинга даты задачи в AddTask: %s", err.Error())
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
			return model.AddTaskResponse{}, fmt.Errorf("ошибка парсинга даты задачи в AddTask: %w", err)
```

Reply:
Нужно везде поправить

Reply:
Еще хорошо бы на английском все писать